project4/3700crawler

```diff
@@ -69,17 +84,30 @@ class WebCrawler:
         path = '/' if not url.path else url.path
         if url.query:
             path += '?' + url.query
-       
-        # Build the request
-        msg = 'GET ' + path + ' HTTP/1.1\nHost: ' + host
-        if self.cookie:
-            msg += 'Cookie: ' + self.cookie
-        msg += CRLF
 
         # Send the request
         sock = self.connect(host)
-        sock.send(str.encode(msg))
-              
+        sock.send(str.encode(make_http('GET', path, {'cookie': self.cookie, 'host':host})))
+
+        # Get the response
+        resp = self.get_response(sock)
+
+        return resp
+
+    def POST(self, url, body):
+        """Implements the HTTP POST method"""
+        # Parse the URL and retrieve the necessary fields
+        url = urlparse(url)
+        host = url.netloc
+        path = '/' if not url.path else url.path
+        if url.query:
+            path += '?' + url.query
+
+        # Send the request
+        sock = self.connect(host)
+        headers = {'cookie': self.cookie, 'host': host, 'content-length': str(len(body)), 'content-type': 'application/x-www-form-urlencoded'}
```

Comment:
I would reformat this to:
```
headers = {
    'cookie': self.cookie,
    'host': host,
    'content-length': str(len(body)),
    'content-type': 'application/x-www-form-urlencoded'
}
```

so that it's easier to read and b/c I've been trying to wrap the lines at 80 characters.